File: catalyst_voices_packages/catalyst_cardano/catalyst_cardano_web/assets/js/catalyst_cardano.js
// Returns available wallet extensions exposed under
// cardano.{walletName} according to CIP-30 standard.
function _getWallets() {
    const cardano = window.cardano;
    if (cardano) {
        const keys = Object.keys(window.cardano);
        const possibleWallets = keys.map((k) => cardano[k]);
        return possibleWallets.filter((w) => typeof w === "object" && "enable" in w && "apiVersion" in w);
    }

    return [];
}

// Returns an instance of `undefined` to dart layer as `undefined`
// cannot be constructed directly in dart layer. Dart nulls are translated
// to JS nulls and JS distinguishes between undefined and null.
function _makeUndefined() {
    return undefined;
}

// A namespace containing the JS functions that
// can be executed from dart side
const catalyst_cardano = {
    getWallets: _getWallets,
    makeUndefined: _makeUndefined,
}

// Expose cardano multiplatform as globally accessible
// so that we can call it via catalyst_cardano.function_name() from
// other scripts or dart without needing to care about module imports
window.catalyst_cardano = catalyst_cardano;

File: catalyst_voices/packages/libs/catalyst_cardano/catalyst_cardano/lib/src/catalyst_cardano.dart
import 'package:catalyst_cardano_platform_interface/catalyst_cardano_platform_interface.dart';

/// A Flutter plugin exposing the [CIP-30](https://cips.cardano.org/cip/CIP-30)
/// and [CIP-95](https://cips.cardano.org/cip/CIP-95) APIs.
final class CatalystCardano {
  /// The default instance of [CatalystCardano] to use.
  static final CatalystCardano instance = CatalystCardano._();

  CatalystCardano._();

  /// Returns available wallet extensions exposed under
  /// cardano.{walletName} according to CIP-30 standard.
  Future<List<CardanoWallet>> getWallets() async {
    return CatalystCardanoPlatform.instance.getWallets();
  }
}
